maskrcnn_benchmark/engine/inference.py

```diff
@@ -386,46 +423,54 @@ def inference(
+            logger.info(res)
+            check_expected_results(res, expected_results, expected_results_sigma_tol)
+            if output_folder:
+                torch.save(res, os.path.join(output_folder, "box_proposals.pth"))
+            return
+        logger.info("Preparing results for COCO format")
+        coco_results = {}
+        if "bbox" in iou_types:
+            logger.info("Preparing bbox results")
+            coco_results["bbox"] = prepare_for_coco_detection(predictions, dataset)
+        if "segm" in iou_types:
+            logger.info("Preparing segm results")
+            coco_results["segm"] = prepare_for_coco_segmentation(predictions, dataset)
+
+        results = COCOResults(*iou_types)
+        logger.info("Evaluating predictions")
+        for iou_type in iou_types:
+            with tempfile.NamedTemporaryFile() as f:
+                file_path = f.name
+                if output_folder:
+                    file_path = os.path.join(output_folder, iou_type + ".json")
+                res = evaluate_predictions_on_coco(
+                    dataset.coco, coco_results[iou_type], file_path, iou_type
                 )
-                key = "AR{}@{:d}".format(suffix, limit)
-                res.results["box_proposal"][key] = stats["ar"].item()
-        logger.info(res)
-        check_expected_results(res, expected_results, expected_results_sigma_tol)
+                results.update(res)
+        logger.info(results)
+        check_expected_results(results, expected_results, expected_results_sigma_tol)
         if output_folder:
-            torch.save(res, os.path.join(output_folder, "box_proposals.pth"))
-        return
-    logger.info("Preparing results for COCO format")
-    coco_results = {}
-    if "bbox" in iou_types:
-        logger.info("Preparing bbox results")
-        coco_results["bbox"] = prepare_for_coco_detection(predictions, dataset)
-    if "segm" in iou_types:
-        logger.info("Preparing segm results")
-        coco_results["segm"] = prepare_for_coco_segmentation(predictions, dataset)
-
-    results = COCOResults(*iou_types)
-    logger.info("Evaluating predictions")
-    for iou_type in iou_types:
-        with tempfile.NamedTemporaryFile() as f:
-            file_path = f.name
-            if output_folder:
-                file_path = os.path.join(output_folder, iou_type + ".json")
-            res = evaluate_predictions_on_coco(
-                dataset.coco, coco_results[iou_type], file_path, iou_type
-            )
-            results.update(res)
-    logger.info(results)
-    check_expected_results(results, expected_results, expected_results_sigma_tol)
-    if output_folder:
-        torch.save(results, os.path.join(output_folder, "coco_results.pth"))
-        
-    return results, coco_results, predictions
-
+            torch.save(results, os.path.join(output_folder, "coco_results.pth"))
+        return results, coco_results
+    if 'coco' in cfg.DATASETS.TEST[0]:
```

Comment:
This doesn't work if we pass multiple datasets during evaluation. I think it might be better to pass a dataset name as an argument to the `inference` function